### PR TITLE
fix: invoke action-setup-tools at lint/test time

### PR DESCRIPTION
**Pull request by dgschwindturo**

### CODE REVIEW

File: auth/action.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: terraform Authentication
+description: Authenticates with terraform
+inputs:
+  checkout-repo:
```
Comment (shakefu) [inline auth/action.yaml:4]: I don't think we need checkout repo here at all since this doesn't work with repo objects.

File: test/action.yaml [hunk 2]
```diff
@@ -59,6 +53,6 @@ runs:
         AWS_SECRET_ACCESS_KEY: ${{ inputs.aws-secret-access-key }}
         TF_CLI_CONFIG_FILE: ${{ inputs.terraform-cli-config-file }}
     - name: Post run
-      if: always() && env._ACTIONS_TF_TEST_CREDENTIALS
+      if: always() && $ {{ steps.create-credentials-file.outputs.terraform-cli-config-file-created }}
```
Comment (shakefu) [inline test/action.yaml:56]: Oof I forgot about this... :/ 

Stupid composites not allowing post-run hooks. This being super-specific to the auth action makes me think it should have a `actions-tf/auth/post-run` or `actions-tf/auth/cleanup` that wraps it and takes no inputs (e.g. instead it reads the `${{ env... }}`? Dunno if that makes sense.

**Review by shakefu — CHANGES_REQUESTED** [review]

File: auth/action.yaml (hunk 1 above)
Comment (dgschwindturo) [reply auth/action.yaml:4]: Wouldn't you expect the `.terraformrc` file to be under git control if it already exists?

Comment (dgschwindturo) [reply auth/action.yaml]: Ah, I misunderstood. I have removed the checkout-repo input and optional Checkout step.

File: test/action.yaml (hunk 2 above)
Comment (dgschwindturo) [reply test/action.yaml:55]: Going to leave this in place for now as we discussed.

**Review by shakefu — APPROVED** [review]

**Pull request merged**